src/CVRF/cvrf_priv.h

```diff
@@ -0,0 +1,354 @@
+struct cvrf_model *cvrf_model_parse_xml(struct oscap_source *source);
+
+/**
+ * Parse CVRF model
+ * @param reader XML Text Reader representing XML model
+ * @return parsed CVRF model
+ */
+struct cvrf_model *cvrf_model_parse(xmlTextReaderPtr reader);
+
+/**
+ * Parse CVRF DocumentPublisher
+ * @param reader XML Text Reader representing XML model
+ * @return parsed CVRF DocumentPublisher
+ */
+struct cvrf_doc_publisher *cvrf_doc_publisher_parse(xmlTextReaderPtr reader);
+
+/**
+ * Parse CVRF DocumentTracking
+ * @param reader XML Text Reader representing XML model
+ * @return parsed CVRF DocumentTracking
+ */
+struct cvrf_doc_tracking *cvrf_doc_tracking_parse(xmlTextReaderPtr reader);
+
+/**
+ * Parse CVRF product tree
+ * @param reader XML Text Reader representing XML model
+ * @return parsed CVRF product tree
+ */
+struct cvrf_product_tree *cvrf_product_tree_parse(xmlTextReaderPtr reader);
+
+/**
+ * Parse CVRF branch
+ * @param reader XML Text Reader representing XML model
+ * @return parsed CVRF branch
+ */
+struct cvrf_branch *cvrf_branch_parse(xmlTextReaderPtr reader);
+
+/**
+ * Parse CVRF Relationship item
+ * @param reader XML Text Reader representing XML model
+ * @return parsed CVRF Relationship
+ */
+struct cvrf_relationship *cvrf_relationship_parse(xmlTextReaderPtr reader);
+
+/**
+ * Parse CVRF vulnerability
+ * @param reader XML Text Reader representing XML model
+ * @return parsed CVRF vulnerability
+ */
+struct cvrf_vulnerability *cvrf_vulnerability_parse(xmlTextReaderPtr reader);
+
+/**
+ * Parse CVRF Remediation
+ * @param reader XML Text Reader representing XML model
+ * @return parsed CVRF Remediation
+ */
+struct cvrf_remediation *cvrf_remediation_parse(xmlTextReaderPtr reader);
+
+/**
+ * Parse Product Status within CVRF vulnerability
+ * @param reader XML Text Reader representing XML model
+ * @return parsed CVRF Product Status
+ */
+struct cvrf_product_status *cvrf_product_status_parse(xmlTextReaderPtr reader);
+
+
+
+/**
+ *
+ * @param index CVRF index with all models to export
+ * @param file OSCAP export target
+ */
+void cvrf_index_export_xml(struct cvrf_index *index, struct oscap_source *export_source);
+
+/**
+ *
+ * @param cvrf CVRF model to export
+ * @param file OSCAP export target
+ */
+void cvrf_model_export_xml(struct cvrf_model *cvrf, struct oscap_source *export_source);
+
+/**
+ *
+ * @param publisher CVRF DocumentPublisher structure to export
+ * @param writer XML Text Writer representing XML model
+ */
+void cvrf_doc_publisher_export(const struct cvrf_doc_publisher *publisher, xmlTextWriterPtr writer);
+
+/**
+ *
+ * @param tracking CVRF DocumentTracking element to export
+ * @param writer XML Text Writer representing XML model
+ */
+void cvrf_doc_tracking_export(const struct cvrf_doc_tracking *tracking, xmlTextWriterPtr writer);
+
+/**
+ *
+ * @param cvrf CVRF model
+ * @param writer XML Text Writer representing XML model
+ */
+void cvrf_export(const struct cvrf_model *cvrf, xmlTextWriterPtr writer);
```

Comment:
Fixed, The export functions were re-written with DOM instead of xmlTextWriterPtr